pkg/logger/handler_test.go

```diff
@@ -78,7 +78,69 @@ func TestLogger(t *testing.T) {
 
 	StartDispatcher(5, logger)
 	httpProxy := httputil.NewSingleHostReverseProxy(targetUri)
-	oh := New(logSvcUrl, sourceUri, v1beta1.LogAll, "mymodel", "default", "default", "default", httpProxy)
+	oh := New(logSvcUrl, sourceUri, v1beta1.LogAll, "mymodel", "default", "default", "default", httpProxy, nil)
+
+	oh.ServeHTTP(w, r)
+
+	b2, _ := io.ReadAll(w.Result().Body)
+	g.Expect(b2).To(gomega.Equal(predictorResponse))
+	// get logRequest
+	<-responseChan
+	// get logResponse
+	<-responseChan
+}
+
+func TestLoggerWithMetadata(t *testing.T) {
+
+	g := gomega.NewGomegaWithT(t)
+
+	predictorRequest := []byte(`{"instances":[[0,0,0]]}`)
+	predictorResponse := []byte(`{"instances":[[4,5,6]]}`)
+
+	responseChan := make(chan string)
+	// Start a local HTTP server
+	logSvc := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		b, err := io.ReadAll(req.Body)
+		g.Expect(err).To(gomega.BeNil())
+		println(string(b))
+		responseChan <- string(b)
+		g.Expect(b).To(gomega.Or(gomega.Equal(predictorRequest), gomega.Equal(predictorResponse)))
+		_, err = rw.Write([]byte(`ok`))
+		g.Expect(err).To(gomega.BeNil())
+		println(req.Header)
+		g.Expect(req.Header["foo"]).To(gomega.Equal([]string{"bar"}))
+
+	}))
+	// Close the server when test finishes
+	defer logSvc.Close()
+
+	// Start a local HTTP server
+	predictor := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		b, err := io.ReadAll(req.Body)
+		g.Expect(err).To(gomega.BeNil())
+		g.Expect(b).To(gomega.Or(gomega.Equal(predictorRequest), gomega.Equal(predictorResponse)))
+		_, err = rw.Write(predictorResponse)
+		g.Expect(err).To(gomega.BeNil())
+	}))
+	// Close the server when test finishes
+	defer predictor.Close()
+
+	reader := bytes.NewReader(predictorRequest)
+	r := httptest.NewRequest("POST", "http://a", reader)
+	r.Header.Add("my-meta/foo", "bar")
```

Comment:
how does this test case validate that this header is passed and other headers not in the allowlist will be filtered?